pandas/tests/tslibs/test_timezones.py

```diff
@@ -118,3 +119,22 @@ def test_maybe_get_tz_invalid_types():
     msg = "<class 'pandas._libs.tslibs.timestamps.Timestamp'>"
     with pytest.raises(TypeError, match=msg):
         timezones.maybe_get_tz(Timestamp.now("UTC"))
+
+
+def test_astype_tz_conversion_roundtrip():
```

Comment:
Can you move these tests to `pandas/tests/frame/methods/test_astype.py`?

And include a ` # GH 35973` comment below

Reply:
done